server-common/src/main/java/org/apache/kafka/admin/AdminUtils.java

```diff
@@ -166,7 +166,7 @@ private static Map<Integer, List<Integer>> assignReplicasToBrokersRackAware(int
                                                                                 int fixedStartIndex,
                                                                                 int startPartitionId) {
         Map<Integer, String> brokerRackMap = new HashMap<>();
-        brokerMetadatas.forEach(m -> brokerRackMap.put(m.id, m.rack.get()));
+        brokerMetadatas.forEach(m -> brokerRackMap.put(m.id, m.rack.orElse(null)));
```

Comment:
Is it expected that `rack` will be empty

Reply:
Before calling this function, the `brokerMetadatas` parameter has already been checked. If `rack` is empty, an `AdminOperationException` will be thrown. However, my IDE still issues a warning, so I made an equivalent replacement. Next time, I will ignore this situation.

Reply:
> Before calling this function, the parameter brokerMetadatas has already been checked. If rack is empty, an AdminOperationException will be thrown. However, my IDE still issues a warning, so I made an equivalent replacement. Next time, I will ignore this situation.

How about moving the check into `assignReplicasToBrokersRackAware`? for example:
```java
brokerMetadatas.forEach(m -> brokerRackMap.put(m.id, m.rack.orElseThrow(() -> new AdminOperationException("Not all brokers have rack information for replica rack aware assignment."))));
```

Reply:
> > Before calling this function, the parameter brokerMetadatas has already been checked. If rack is empty, an AdminOperationException will be thrown. However, my IDE still issues a warning, so I made an equivalent replacement. Next time, I will ignore this situation.
> 
> How about moving the check into `assignReplicasToBrokersRackAware`? for example:
> 
> ```java
> brokerMetadatas.forEach(m -> brokerRackMap.put(m.id, m.rack.orElseThrow(() -> new AdminOperationException("Not all brokers have rack information for replica rack aware assignment."))));
> ```

`great suggestion, I will do it`